logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/message/ApplicationMessageHandler.kt

```diff
@@ -204,42 +212,33 @@ internal class ApplicationMessageHandlerImpl(
 
         when (val content = message.content) {
             // Persist Messages - > lists
-            is MessageContent.Text -> handleTextMessage(message, content)
-
-            is MessageContent.FailedDecryption -> {
-                persistMessage(message)
-            }
-
-            is MessageContent.Knock -> handleKnock(message)
+            is MessageContent.Text -> persistMessage(adjustTextMessageWithQoutedReference(message, content))
+            is MessageContent.FailedDecryption -> persistMessage(message)
+            is MessageContent.Knock -> persistMessage(message)
             is MessageContent.Asset -> assetMessageHandler.handle(message)
-
+            is MessageContent.RestrictedAsset -> TODO()
             is MessageContent.Unknown -> {
                 logger.i(message = "Unknown Message received: { \"message\" : ${message.toLogString()} }")
                 persistMessage(message)
             }
 
-            is MessageContent.RestrictedAsset -> TODO()
         }
     }
 
-    private suspend fun handleKnock(message: Message.Regular) {
-        persistMessage(message)
-    }
-
-    private suspend fun handleTextMessage(
+    private suspend fun adjustTextMessageWithQoutedReference(
```

Comment:
```suggestion
    private suspend fun adjustTextMessageWithQuotedReference(
```

As seen in [here](https://github.com/wireapp/kalium/pull/1552/files/e89bcf5c5cb6ed668eb3aae40254d0243059eac1#r1132345115).

Reply:
To be honest. I preferred the old name.

From the caller (`processMessage`) point of view, it is just calling `handleTextMessage`. It doesn't need to know WHAT in a text message should be processed.

By changing this name, now `processMessage` needs to understand how the text message should be processed, instead of just delegating to specialised function.

If clarity was the reason for changing this name, I'd prefer a simple comment inside the function, or a nested function like so:

```kotlin
    private suspend fun handleTextMessage(
        message: Message.Regular,
        messageContent: MessageContent.Text
    ): Message.Regular {
        fun adjustedQuoteReference(): QuoteReference {
            val quotedReference = messageContent.quotedMessageReference
            return if (quotedReference != null) {
                verifyMessageQuote(quotedReference, message)
            } else {
                messageContent.quotedMessageReference
            }
        }
        return message.copy(
            content = messageContent.copy(quotedMessageReference = adjustedQuoteReference())
        )
    }
```